include/zephyr/drivers/regulator.h

```diff
@@ -544,6 +555,52 @@ static inline int regulator_get_mode(const struct device *dev,
 	return api->get_mode(dev, mode);
 }
 
+/**
+ * @brief Set active discharge setting.
+ *
+ * @param dev Regulator device instance.
+ * @param active_discharge Active discharge enable or disable.
+ *
+ * @retval 0 If successful.
+ * @retval -ENOSYS If function is not implemented.
+ * @retval -errno In case of any other error.
+ */
+static inline int regulator_set_active_discharge(const struct device *dev,
+				     uint8_t active_discharge)
```

Comment:
```suggestion
				     bool enable)
```

Reply:
my concern with making it a bool would be handle the issue of it not being defined where it just gets set to the max value of a UINT8

Reply:
I think bool would be better if we can assume that active_discharge is always disabled by default.  If this is not a safe assumption I think it is OK like this.

Reply:
From an API client perspective, this function will always be called with either `0` or `1` (bool), am I wrong?

Reply:
yeah. I updated to be a bool